utils/s3/credentials_providers/instance_profile_credentials_provider.cc

```diff
@@ -34,7 +34,7 @@ future<> instance_profile_credentials_provider::reload() {
 
 future<> instance_profile_credentials_provider::update_credentials() {
     auto factory = std::make_unique<utils::http::dns_connection_factory>(ec2_metadata_ip, port, false, ec2_md_logger);
-    http::experimental::client http_client(std::move(factory), 1, http::experimental::client::retry_requests::yes);
+    retryable_http_client http_client(std::move(factory), 1, retryable_http_client::ignore_exception, http::experimental::client::retry_requests::yes, retry_strategy);
```

Comment:
Nitpick on the commit message: I would use "s3", not "credentials_provider", in the prefix in the title. It's a feature specific to the s3 feature, it's not some sort of general "credentials provider" used throughout Scylla.

Reply:
it is not S3 specific but AWS, will change it.

Reply:
If I understand correctly, we currently call this feature "s3" - we have utils/s3, not utils/aws. I meant "s3" in the sense that this is for the s3 feature - even if the s3 feature ends up calling other AWS services, not just s3.

Anyway, I already merged this PR so you don't need to create new versions. If there's something you want to change (but not a commit message...) please send a new followup PR.